src/components/signInForm.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import Button from 'react-bootstrap/Button';
+
+import { Formik, Field, Form, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
+
+function SignInForm (props){
+    const onSubmit = props.onSubmit
+    const mutationError = props.mutationError
+
+    return(
+
+        <Formik
+            initialValues={{ username: '', password: ''}}
+            validationSchema={Yup.object({
+                username: Yup.string()
+                .min(3, 'Must be 3 characters or more')
+                .max(20, 'Must be 20 characters or less')
+                .required('Required'),
+                password: Yup.string()
+                .min(8, 'Must be 8 characters or more')
+                .max(30, 'Must be 30 characters or less')
+                .required('Required'),
+            })}
```

Comment:
Good! It would be even better if you extract that Yup object to make the Formik component easier to read.